packages/firestore/src/local/indexeddb_index_manager.ts

```diff
@@ -31,30 +30,17 @@ import { SimpleDbStore } from './simple_db';
  * A persisted implementation of IndexManager.
  */
 export class IndexedDbIndexManager implements IndexManager {
-  /**
```

Comment:
I can't find where this cache is kept now. Unconditionally writing it would be a pretty major regression.

Or is the equivalent supposed to be the `collectionParents` addition in `indexeddb_mutation_queue.ts`? This cache ensures that we only ever write a collection parent index entry once for the entire duration of the app. The `indexeddb_mutation_queue.ts` changes ensure uniqueness within a mutation batch, but that's a far lesser benefit.

On the whole I think unwinding all of this in the name of idempotency is a bad deal. If we added `transaction.onCommitted` this wouldn't be impacted at all. This is the second case we were looking for--I think it makes sense to pursue that instead of these changes.